doc/tuned-adm.1
.\"/* 
.\" * All rights reserved
.\" * Copyright (C) 2009-2014 Red Hat, Inc.
.\" * Authors: Marcela Mašláňová, Phil Knirsch, Jaroslav Škarvada
.\" *
.\" * This program is free software; you can redistribute it and/or
.\" * modify it under the terms of the GNU General Public License
.\" * as published by the Free Software Foundation; either version 2
.\" * of the License, or (at your option) any later version.
.\" *
.\" * This program is distributed in the hope that it will be useful,
.\" * but WITHOUT ANY WARRANTY; without even the implied warranty of
.\" * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
.\" * GNU General Public License for more details.
.\" *
.\" * You should have received a copy of the GNU General Public License
.\" * along with this program; if not, write to the Free Software
.\" * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
.\" */
.\" 
.TH TUNED_ADM "1" "15 Oct 2014" "Linux Programmer's Manual"
.SH NAME
tuned\-adm - commandline tool for switching between different tuning profiles
.SH SYNOPSIS
.B tuned\-adm 
.B 
.BR [ " help ", " list ", " active ", " profile <profile> ", " off " ]
.br
.SH DESCRIPTION
This command line utility allows user to switch between user definable tuning profiles. Several predefined profiles are already included for some of the more common cases. Although the profiles typically refer to a common use case you can of course switch to any profile on any system. So switching to the laptop-battery-powersave profile on a server will activate this profile on it. This way, depending on your requirements you can select any of the profiles that fits your needs. You can even create your own profile, either based on one of the existing ones by copying it or make a completely new one. The profiles are stored in subdirectories below /etc/tune-profiles.

.SH "OPTIONS"

.SS
.TP
.B help
Print out the list of commands. It's also the default if you start tuned\-adm without any commands.

.TP
.B list
List all available profiles.

.TP
.B active
Show current active profile.

.TP
.B profile <profile>
Switches to the given profilename. If none is given or no valid one is given the command gracefully exits without performing any operation.

.TP
.B off
Switch off all ktune and tuned settings and remove them from services. The configuration file are set to previous settings.

.SH "FILES"
.nf
/etc/tune\-profiles/*

.SH "SEE ALSO"
.BR tuned (8)
.BR tuned\-profiles (7)
.BR tuned\-profiles\-sap (7)
.BR tuned\-profiles\-sap\-hana (7)
.SH AUTHOR
.NF
Marcela Mašláňová <mmaslano@redhat.com>
Phil Knirsch <pknirsch@redhat.com>
Jaroslav Škarvada <jskarvad@redhat.com>


ChangeLog
[33mcommit adf6d855789f1e5738ae0a113a96d914b480e23f[m
Author: Jan Vcelak <jvcelak@redhat.com>
Date:   Tue Jan 29 16:05:50 2013 +0100

    udev rules: ask tuned to restart ktune instead of restarting it directly

[33mcommit afd26f62774daf81a9f95d256d415547300f7757[m
Author: Jan Vcelak <jvcelak@redhat.com>
Date:   Tue Jan 29 13:26:25 2013 +0100

    tuned: restart ktune 10 seconds after last SIGUSR2 was received
    
    When multiple devices are added into the system, udev restarted ktune
    daemon for each new device. This can cause races within ktune. This
    mechanism allows triggering just one ktune restart when multiple
    devices are inserted.

[33mcommit 6fc759a9fff5801a9f80e672207fc41f95b3ae7b[m
Author: Jan Vcelak <jvcelak@redhat.com>
Date:   Tue Jan 29 14:32:30 2013 +0100

    functions: implement support for readahead restore

[33mcommit f46c9d05477f93f31e3c5ee1c24637f4adde743c[m
Author: Jaroslav Škarvada <jskarvad@redhat.com>
Date:   Thu Jan 3 17:03:43 2013 +0100

    profiles: removed the nobarrier from the virtual-guest profile
    
    resolves: #886956

[33mcommit 355fd87b1b2304863fe04266e290dc99394587ad[m
Author: Jaroslav Škarvada <jskarvad@redhat.com>
Date:   Thu Jan 3 17:01:48 2013 +0100

    disabled transparent hugepages in latency-performance profile
    
    resolves: rhbz#887355

[33mcommit 66d8e396db3845463abc181c541ee09c686b1ff1[m
Author: Jan Vcelak <jvcelak@redhat.com>
Date:   Fri Dec 7 14:06:09 2012 +0100

    {disk,net}devstat: fix typo in usage

[33mcommit c1b7f35546a23b409ca0e1479d928b8d7f57c06b[m
Author: Jan Vcelak <jvcelak@redhat.com>
Date:   Wed Oct 10 10:47:26 2012 +0200

    udev: add rule to restart ktune in new block device is added
    
    Resolves: #847445

[33mcommit 64b45e8a04b06c0f554c64b458c590c9904f7f5f[m
Author: Jan Vcelak <jvcelak@redhat.com>
Date:   Thu Sep 27 13:40:38 2012 +0200

    profiles: do not disable barriers on devices with write back cache
    
    Resolves: #801561

[33mcommit 9e8f67009568adce6b9a2f1d4c4c828b2192cba8[m
Author: Jan Vcelak <jvcelak@redhat.com>
Date:   Mon Sep 24 15:08:19 2012 +0200

    tuned: create PID file with proper permissions
    
    Resolves: #845336

[33mcommit 84536e5a5917282821a654cb5d47ac3505d9506e[m
Author: Jan Vcelak <jvcelak@redhat.com>
Date:   Mon Sep 24 14:51:59 2012 +0200

    ktune: validate value of ELEVATOR_TUNE_DEVS
    
    Resolves: #784308

[33mcommit d4c0b5a2d145da188b995419764f9c969b9afdbb[m
Author: Jan Vcelak <jvcelak@redhat.com>
Date:   Tue Mar 20 14:29:04 2012 +0100

    update systemd unit file install path
    
    required by /lib -> /usr/lib feature

[33mcommit 9dfa60a1f4166aab80e400d88a3ee28bcc256efc[m
Author: Jan Vcelak <jvcelak@redhat.com>
Date:   Tue Mar 20 14:16:42 2012 +0100

    spec: missing system-units build requirement

[33mcommit cbb2bf6e85117a0f245de5010e70c75a8ccc35b6[m
Author: Jan Vcelak <jvcelak@redhat.com>
Date:   Tue Mar 20 13:55:48 2012 +0100

    version bump

[33mcommit 0427cea0d801b03ab5fbfdeb3c26922c6eaa2247[m
Author: Jan Vcelak <jvcelak@redhat.com>
Date:   Mon Mar 5 08:51:53 2012 +0100

    netdevstat: cmdline error handling

[33mcommit 81590cf7705e982c7880c69d01ea6d02a6e7a2e4[m
Author: Jan Vcelak <jvcelak@redhat.com>
Date:   Mon Mar 5 08:45:50 2012 +0100

    diskdevstat: cmdline error handling

[33mcommit 3b10ac690bc6164b721236702cb5eb8493278f1e[m
Author: Jan Vcelak <jvcelak@redhat.com>
Date:   Fri Mar 2 09:59:29 2012 +0100

    virtual-guest: cleanup ktune.sh
    
    References: rhbz#740976, rhbz#740977

[33mcommit 90afd92add2facb02596a7e8e1341f252ae3684d[m
Author: Jan Vcelak <jvcelak@redhat.com>
Date:   Fri Mar 2 09:00:26 2012 +0100

    virtual-host: tune virtual memory settings
    
    References: rhbz#740976

[33mcommit 78012939d465405587a6607838f15e8c06cbd5a3[m
Author: Jan Vcelak <jvcelak@redhat.com>
Date:   Wed Jan 11 13:38:28 2012 +0100

    tuned-adm: sort list of available profiles

[33mcommit e30a77ede22df20232547b6243107b57ed0ad0ee[m
Author: Jan Vcelak <jvcelak@redhat.com>
Date:   Wed Jan 11 13:35:45 2012 +0100

    tuned-adm man: add virtual-host profile decription

[33mcommit 5984ad2de7e2e5ee76d54191f03515ae55739a4e[m
Author: Jan Vcelak <jvcelak@redhat.com>
Date:   Wed Jan 11 13:26:31 2012 +0100

    add virtual-host profile

[33mcommit 0e4826f78b14f219c876bc9d3b35834cd7a18311[m
Author: Jan Vcelak <jvcelak@redhat.com>
Date:   Wed Jan 11 12:59:48 2012 +0100

    tuned-adm: fix crash when active-profile file is missing

[33mcommit 0754d462ff72ed69b704b3a4a4d7a5ecc77df655[m
Author: Jan Vcelak <jvcelak@redhat.com>
Date:   Wed Jan 11 12:55:22 2012 +0100

    tuned-adm man: add virtual-guest profile decription

[33mcommit 8bb2e57c523cc27915b422d077ba5c644bd85f6a[m
Author: Jan Vcelak <jvcelak@redhat.com>
Date:   Wed Jan 11 12:48:19 2012 +0100

    add virtual-guest profile

[33mcommit 8bd483a47c4e00cf106fb16d50ac5d0c61f21ab8[m
Author: Jan Vcelak <jvcelak@redhat.com>
Date:   Wed Jan 11 12:40:23 2012 +0100

    functions: support for disk scheduler quantum setting

[33mcommit b5297bcc8cdb71b7c7fff17b7a18ea194952cc1c[m
Author: Jan Vcelak <jvcelak@redhat.com>
Date:   Wed Jan 11 12:39:47 2012 +0100

    functions: update disks detection mechanism

[33mcommit f11bae54bc26e2969ab1413995baeae472e75b60[m
Author: Jaroslav Škarvada <jskarvad@redhat.com>
Date:   Sat Dec 3 15:31:29 2011 +0100

    set_radeon_powersave: fixed profile name

[33mcommit 229b5f7874c406f702b7f1fd9d4e18a1df0e575e[m
Author: Jaroslav Škarvada <jskarvad@redhat.com>
Date:   Sat Dec 3 15:14:06 2011 +0100

    Changed Radeon PM to auto in laptop-battery-powersave
    
    It should be safer for presenters, because the more
    aggresive dynpm doesn't support multi heads.

[33mcommit 60f78e3dc83907d558913473acc145ce864d4c26[m
Author: Jaroslav Škarvada <jskarvad@redhat.com>
Date:   Sat Dec 3 14:32:39 2011 +0100

    Improved radeon powersavings settings in tuned profiles

[33mcommit 9280e4ee158792973112ea78c3848bc25609c51e[m

tuned-adm.py
#!/usr/bin/python -Es
#
# tuned: daemon for monitoring and adaptive tuning of system devices
#
# Copyright (C) 2008-2013 Red Hat, Inc.
#
# This program is free software; you can redistribute it and/or
# modify it under the terms of the GNU General Public License
# as published by the Free Software Foundation; either version 2
# of the License, or (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with this program; if not, write to the Free Software
# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
#

import argparse
import sys
import traceback
import tuned.admin
import tuned.consts as consts
import tuned.version as ver

if __name__ == "__main__":
	parser = argparse.ArgumentParser(description="Manage tuned daemon.")
	parser.add_argument('--version', "-v", action = "version", version = "%%(prog)s %s.%s.%s" % (ver.TUNED_VERSION_MAJOR, ver.TUNED_VERSION_MINOR, ver.TUNED_VERSION_PATCH))
	parser.add_argument("--debug", "-d", action="store_true", help="show debug messages")
	subparsers = parser.add_subparsers()

	parser_list = subparsers.add_parser("list", help="list available profiles")
	parser_list.set_defaults(action="list")

	parser_active = subparsers.add_parser("active", help="show active profile")
	parser_active.set_defaults(action="active")

	parser_off = subparsers.add_parser("off", help="switch off all tunings")
	parser_off.set_defaults(action="off")

	parser_profile = subparsers.add_parser("profile", help="switch to a given profile")
	parser_profile.set_defaults(action="profile")
	parser_profile.add_argument("profiles", metavar="profile", type=str, nargs="+", help="profile name")

	parser_off = subparsers.add_parser("recommend", help="recommend profile")
	parser_off.set_defaults(action="recommend_profile")

	args = parser.parse_args(sys.argv[1:])

	options = vars(args)
	debug = options.pop("debug")
	action_name = options.pop("action")
	result = False

	try:
		controller = tuned.admin.DBusController(consts.DBUS_BUS, consts.DBUS_OBJECT, consts.DBUS_INTERFACE, debug)
		admin = tuned.admin.Admin(controller, debug)

		action = getattr(admin, action_name)
		result = action(**options)
	except tuned.admin.TunedAdminException as e:
		if not debug:
			print >>sys.stderr, e
		else:
			traceback.print_exc()
		sys.exit(2)
	except:
		traceback.print_exc()
		sys.exit(3)

	if result == False:
		sys.exit(1)
	else:
		sys.exit(0)
